src/models/miner.model.js

```diff
@@ -40,19 +63,142 @@ class MinerModel {
      */
     getJob() {
         return new Promise((resolve, reject) => {
+            
+
+
+
+
             resolve('Boilerplate for Miner::getJob');
         })
     }
-
     /**
-     * @param {BlockTemplate} blockTemplate 
-     */
-    submit(blockTemplate) {
+     * 
+     * @Example Request
+     * "id": 2,
+     * "jsonrpc": "2.0",
+     * "method": "submit",
+     * "params": {
+     * "id": "1be0b7b6-b15a-47be-a17d-46b2911cf7d0",
+     * "job_id": "4BiGm3/RgGQzgkTI/xV0smdA+EGZ",
+     * "nonce": "d0030040",
+     * "result": "e1364b8782719d7683e2ccd3d8f724bc59dfa780a9e960e7c0e0046acdb40100"
+     * }
+     * @param {request} req
+     */ 
+    submit(req) {
         return new Promise((resolve, reject) => {
-            resolve('Boilerplate for Miner::submit');
+            //miner, job, blockTemplate, nonce, resultHash
+            const minerData = 
```

Comment:
the `return new Promise()` was just a place holder, please dont use it like that, many of our Services, Api calls are already returning promises, let'st just chain them instead.